### PR TITLE
chore: Misc. cleanup based on using a fresh Bison app in production.

### PR DESCRIPTION
**Pull request by cullylarson**
## Changes

- chore: Remove unused imports in `jest.teardown.js`.
- chore: Remove commented code block in `jest.setup.js.ejs`.
- fix: `pageProps` in `AllProviders` might not be defined.
- chore: Don't validate `LoginForm` and `SignUpForm` until they're submitted.
- chore: Remove unnecessary isSubmitted checks in `LoginForm` and `SignUpForm`.
- chore: Remove .gitignored e2e storage stage json files.
- chore: Don't truncate the _migration table in resetDB.

## Checklist

- [ ] Requires dependency update?
- [ ] Generating a new app works

### CODE REVIEW

File: packages/create-bison-app/template/components/AllProviders.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ function AppWithAuth({ children }: { children: React.ReactNode }) {
  * Renders all context providers
  */
 export function AllProviders({ pageProps, children }: Props) {
-  const { cookies, session } = pageProps;
+  const { cookies, session } = pageProps || {};
```
Comment (cullylarson) [inline packages/create-bison-app/template/components/AllProviders.tsx:41]: `pageProps` could be `undefined`.

File: packages/create-bison-app/template/components/auth/LoginForm.tsx [hunk 2]
```diff
@@ -25,8 +25,8 @@ export function LoginForm() {
   const {
     register,
     handleSubmit,
-    formState: { errors, isValid, isSubmitted },
-  } = useForm<LoginFormData>({ mode: 'onChange' });
+    formState: { errors, isValid },
+  } = useForm<LoginFormData>();
```
Comment (cullylarson) [inline packages/create-bison-app/template/components/auth/LoginForm.tsx:29]: We don't want to validate until the form is submitted (it's annoying to see an error message while you're typing).

File: packages/create-bison-app/template/components/auth/LoginForm.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ export function LoginForm() {
         </Text>
       </Flex>
       <Stack spacing="5">
-        <FormControl isInvalid={Boolean(isSubmitted && errors.email)}>
+        <FormControl isInvalid={Boolean(errors.email)}>
```
Comment (cullylarson) [inline packages/create-bison-app/template/components/auth/LoginForm.tsx:76]: Including `isSubmitted` in these checks is unnecessary.

File: packages/create-bison-app/template/tests/helpers/db.ts [hunk 4]
```diff
@@ -28,7 +28,7 @@ export const resetDB = async (): Promise<boolean> => {
       (SELECT 'TRUNCATE TABLE ' || string_agg(oid::regclass::text, ', ') || ' CASCADE'
         FROM   pg_class
         WHERE  relkind = 'r'  -- only tables
-        AND    relname != '_Migration'
+        AND    relname NOT LIKE '%_migration%'
```
Comment (cullylarson) [inline packages/create-bison-app/template/tests/helpers/db.ts:31]: Before this change, `resetDB` would remove the `_migration` table.

File: packages/create-bison-app/template/tests/jest.setup.js.ejs [hunk 5]
```diff
@@ -6,13 +6,6 @@ const { nanoid } = require('nanoid');
 const exec = util.promisify(childProcess.exec);
 
 module.exports = async () => {
-  // console.log(`\n[jest] waiting for server's health check...`);
-  // let isUp = false;
-  // while (isUp === false) {
-  //   await new Promise((r) => setTimeout(r, 250));
-  //   isUp = await statusCheck();
-  // }
-  // console.log(`[jest] server is up...`);
```
Comment (cullylarson) [inline packages/create-bison-app/template/tests/jest.setup.js.ejs:15]: Do we ever use this? Are there better ways to wait for the server to start? Also, `statusCheck` isn't defined.

File: packages/create-bison-app/template/tests/jest.teardown.js [hunk 6]
```diff
@@ -1,11 +1,4 @@
-const util = require('util');
```
Comment (cullylarson) [inline packages/create-bison-app/template/tests/jest.teardown.js:1]: This and the other imports removed in this file were unused.

File: packages/create-bison-app/template/tests/utils.tsx [hunk 7]
```diff
@@ -99,6 +99,7 @@ const mockRouter: NextRouter = {
   replace: jest.fn(),
   reload: jest.fn(),
   back: jest.fn(),
+  forward: jest.fn(),
```
Comment (cullylarson) [inline packages/create-bison-app/template/tests/utils.tsx]: This must have been added to next router recently.

File: packages/create-bison-app/template/tests/e2e/temp/adminStorageState.json [hunk 8]
```diff
@@ -1,49 +0,0 @@
-{
```
Comment (cullylarson) [inline packages/create-bison-app/template/tests/e2e/temp/adminStorageState.json:1]: These json files were probably included by accident. They are targeted in `.gitignore`.

**Review by alexanderson1993 — APPROVED** [review]

File: packages/create-bison-app/template/tests/utils.tsx (hunk 7 above)
Comment (cullylarson) [reply packages/create-bison-app/template/tests/utils.tsx]: Removed this. The `forward` prop isn't supported in the version of Bison we're using.

**Pull request merged**